core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1096,6 +1096,9 @@ public T next() {
 
         public Map.Entry<OakRBuffer, OakRBuffer> next() {
             Map.Entry<ByteBuffer, Handle> pair = advance();
+            if (pair.getKey() == null) {
```

Comment:
I remember I got a NPE on a key one day, but I do not remember the exact circumstances. I removed the check, tests passed, hope not to see a NPE again...

Reply:
:) 
It would have been interesting to dive into the root cause of how null keys get stored in the map in the first place. Might be a good idea to back up your hope with an open issue reminding us to look into this.